### PR TITLE
Add Pages logs v2 endpoint

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
Deprecated old endpoint code and added things for the logs v2 endpoint.

## Has your change been tested?
Unit tests pass. Full test provided in https://github.com/cloudflare/cloudflare-go/pull/928#issuecomment-1166651035

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly. -- Waiting for API docs PR to be merged
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs. -- Waiting on PR

### CODE REVIEW

File: pages_deployment.go [hunk 1]
```diff
@@ -181,6 +210,32 @@ func (api *API) GetPagesDeploymentStageLogs(ctx context.Context, params GetPages
...
 }
 
+// GetPagesDeploymentStageLogs returns the logs for a Pages deployment stage.
+//
+// API reference: https://api.cloudflare.com/#pages-deployment-get-deployment-stage-logs
+func (api *API) GetPagesDeploymentLogs(ctx context.Context, params GetPagesDeploymentLogsParams) (PagesDeploymentLogs, error) {
+	v := url.Values{}
+	if params.Size > 0 {
+		v.Set("size", strconv.Itoa(params.Size))
+	}
```
Comment (jacobbednarz) [inline pages_deployment.go:220]: we don't need to manually do this if we use the `url` tags in the structs followed by [`buildURI`](https://github.com/cloudflare/cloudflare-go/blob/e5c6a2d8740459c045a4f020b5510a87b29fa548/utils.go#L10) passing in the values we need here. check out https://github.com/cloudflare/cloudflare-go/blob/master/tunnel_routes.go#L31-L41 for some example usage of this.

File: pages_deployment.go [hunk 2]
```diff
@@ -181,6 +210,32 @@ func (api *API) GetPagesDeploymentStageLogs(ctx context.Context, params GetPages
...
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/deployments/%s/history/logs", params.AccountID, params.ProjectName, params.DeploymentID)
```
Comment (jacobbednarz) [inline pages_deployment.go:222]: you'll want to check that `params.AccountID`, `params.ProjectName`, `params.DeploymentID` are not empty before trying to use them here.

**Comment by jacobbednarz** [comment]
once you confirm this working on your side, i'm happy to land it.

**Comment by WalshyDev** [comment]
Docs are live, updated the link and tested now too. All looks to be working good.

**Test cases**

Test case 1:
<details>
  <summary>Code</summary>
  
  ```go
    package main
    
    import (
	    "context"
	    "log"
	    "os"
    
	    "github.com/cloudflare/cloudflare-go"
    )
    
    func main() {
	    api, err := cloudflare.New(os.Getenv("API_KEY"), os.Getenv("EMAIL"))
	    if err != nil {
		    log.Fatal(err)
	    }
    
	    ctx := context.Background()
    
	    logs, err := api.GetPagesDeploymentLogs(ctx, cloudflare.GetPagesDeploymentLogsParams{
		    AccountID:    os.Getenv("ACCOUNT_TAG"),
		    ProjectName:  "walshys-cool-project",
		    DeploymentID: "ec1b0d9d-b3c8-4c77-b881-45e0c51486dd",
	    })
    
	    if err != nil {
		    log.Fatal(err)
	    }
    
	    for _, line := range logs.Data {
		    log.Println(line.Line)
	    }
    }
  ```
</details>

returns:
<details>
  <summary>Logs</summary>
  
  ```
    2022/06/26 21:48:19 Cloning repository...
    2022/06/26 21:48:19 From https://github.com/WalshyDev/pages-testing
    2022/06/26 21:48:19  * branch            209c5bb11d89533f426b2f8469bcae12fdccf71b -> FETCH_HEAD
    2022/06/26 21:48:19
    2022/06/26 21:48:19 HEAD is now at 209c5bb Update index.html
    2022/06/26 21:48:19
    2022/06/26 21:48:19
    2022/06/26 21:48:19 Success: Finished cloning repository files
    2022/06/26 21:48:19 Installing dependencies
    2022/06/26 21:48:19 Python version set to 2.7
    2022/06/26 21:48:19 v12.18.0 is already installed.
    2022/06/26 21:48:19 Now using node v12.18.0 (npm v6.14.4)
    2022/06/26 21:48:19 Started restoring cached build plugins
    2022/06/26 21:48:19 Finished restoring cached build plugins
    2022/06/26 21:48:19 Attempting ruby version 2.7.1, read from environment
    2022/06/26 21:48:19 Using ruby version 2.7.1
    2022/06/26 21:48:19 Using PHP version 5.6
    2022/06/26 21:48:19 5.2 is already installed.
    2022/06/26 21:48:19 Using Swift version 5.2
    2022/06/26 21:48:19 Installing Hugo 0.54.0
    2022/06/26 21:48:19 Hugo Static Site Generator v0.54.0-B1A82C61A/extended linux/amd64 BuildDate: 2019-02-01T10:04:38Z
    2022/06/26 21:48:19 Started restoring cached go cache
    2022/06/26 21:48:19 Finished restoring cached go cache
    2022/06/26 21:48:19 go version go1.14.4 linux/amd64
    2022/06/26 21:48:19 go version go1.14.4 linux/amd64
    2022/06/26 21:48:19 Installing missing commands
    2022/06/26 21:48:19 Verify run directory
    2022/06/26 21:48:19 Executing user command: printenv;  erl -eval 'erlang:display(erlang:system_info(otp_release)), halt().'  -noshell
    2022/06/26 21:48:19 NVM_INC=/opt/buildhome/.nvm/versions/node/v12.18.0/include/node
    2022/06/26 21:48:19 NODE_VERSION=v12.18.0
    2022/06/26 21:48:19 GIMME_CGO_ENABLED=true
    2022/06/26 21:48:19 WASMER_CACHE_DIR=/opt/buildhome/.wasmer/cache
    2022/06/26 21:48:19 rvm_bin_path=/opt/buildhome/.rvm/bin
    2022/06/26 21:48:19 NF_IMAGE_TAG=v3.4.0
    2022/06/26 21:48:19 WASMER_DIR=/opt/buildhome/.wasmer
    2022/06/26 21:48:19 GEM_HOME=/opt/buildhome/.rvm/gems/ruby-2.7.1
    2022/06/26 21:48:19 NVM_CD_FLAGS=
    2022/06/26 21:48:19 CF_PAGES_BRANCH=main
    2022/06/26 21:48:19 SWIFTENV_ROOT=/opt/buildhome/.swiftenv
    2022/06/26 21:48:19 TERM=unknown
    2022/06/26 21:48:19 SHELL=/bin/bash
    2022/06/26 21:48:19 NF_IMAGE_VERSION=68f5e4d1306f77b1959f42795d4901c8974b4fd8
    2022/06/26 21:48:19 IRBRC=/opt/buildhome/.rvm/rubies/ruby-2.7.1/.irbrc
    2022/06/26 21:48:19 PIPENV_VENV_IN_PROJECT=1
    2022/06/26 21:48:19 GIMME_TYPE=binary
    2022/06/26 21:48:19 NODE_OPTIONS=--max_old_space_size=4096
    2022/06/26 21:48:19 LZ4_VERSION=1.8.0
    2022/06/26 21:48:19 OLDPWD=/opt/buildhome
    2022/06/26 21:48:19 MY_RUBY_HOME=/opt/buildhome/.rvm/rubies/ruby-2.7.1
    2022/06/26 21:48:19 LC_ALL=en_US.UTF-8
    2022/06/26 21:48:19 NVM_DIR=/opt/buildhome/.nvm
    2022/06/26 21:48:19 USER=buildbot
    2022/06/26 21:48:19 CF_PAGES_COMMIT_SHA=209c5bb11d89533f426b2f8469bcae12fdccf71b
    2022/06/26 21:48:19 SUDO_USER=root
    2022/06/26 21:48:19 CUSTOM_RUBY=0
    2022/06/26 21:48:19 SUDO_UID=0
    2022/06/26 21:48:19 rvm_path=/opt/buildhome/.rvm
    2022/06/26 21:48:19 RVM_DIR=/opt/buildhome/.rvm
    2022/06/26 21:48:19 USERNAME=buildbot
    2022/06/26 21:48:19 CF_PAGES_URL=https://ec1b0d9d.walshys-cool-project.pages.dev
    2022/06/26 21:48:19 VIRTUAL_ENV=/opt/buildhome/python2.7
    2022/06/26 21:48:19 GIMME_NO_ENV_ALIAS=true
    2022/06/26 21:48:19 rvm_prefix=/opt/buildhome
    2022/06/26 21:48:19 PATH=/opt/buildhome/.wasmer/bin:/opt/buildhome/.gimme/versions/go1.14.4.linux.amd64/bin:/opt/buildhome/cache/.binrc-a5679f71f5966d1b3450c8dcd52d4743ec08e632/binaries/gohugoio/hugo/v0.54.0:/opt/buildhome/.rvm/gems/ruby-2.7.1/bin:/opt/buildhome/.rvm/gems/ruby-2.7.1@global/bin:/opt/buildhome/.rvm/rubies/ruby-2.7.1/bin:/opt/buildhome/.rvm/bin:/opt/buildhome/.nvm/versions/node/v12.18.0/bin:/opt/buildhome/python2.7/bin:/opt/buildhome/.swiftenv/bin:/opt/buildhome/.swiftenv/shims:/opt/buildhome/.php:/opt/buildhome/.binrc/bin:/usr/local/rvm/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/buildhome/.cask/bin:/opt/buildhome/.gimme/bin:/opt/buildhome/.dotnet/tools:/opt/buildhome/.dotnet:/opt/buildhome/.wasmer/globals/wapm_packages/.bin
    2022/06/26 21:48:19 PIPENV_DEFAULT_PYTHON_VERSION=2.7
    2022/06/26 21:48:19 HUGO_VERSION=0.54.0
    2022/06/26 21:48:19 NVM_VERSION=0.35.3
    2022/06/26 21:48:19 PWD=/opt/buildhome/repo
    2022/06/26 21:48:19 PIPENV_RUNTIME=2.7
    2022/06/26 21:48:19 ELM_VERSION=0.19.0-bugfix6
    2022/06/26 21:48:19 LANG=en_US.UTF-8
    2022/06/26 21:48:19 JAVA_VERSION=default_sdk
    2022/06/26 21:48:19 NETLIFY_BUILD_SWIFT_VERSION=5.2
    2022/06/26 21:48:19 PS1=(python2.7)
    2022/06/26 21:48:19 rvm_version=1.29.10 (latest)
    2022/06/26 21:48:19 CUSTOM_SWIFT=0
    2022/06/26 21:48:19 SUDO_COMMAND=/opt/build/bin/build printenv;  erl -eval 'erlang:display(erlang:system_info(otp_release)), halt().'  -noshell
    2022/06/26 21:48:19 SHLVL=3
    2022/06/26 21:48:19 HOME=/opt/buildhome
    2022/06/26 21:48:19 GOROOT=/opt/buildhome/.gimme/versions/go1.14.4.linux.amd64
    2022/06/26 21:48:19 CF_PAGES=1
    2022/06/26 21:48:19 LANGUAGE=en_US:en
    2022/06/26 21:48:19 YARN_VERSION=1.22.4
    2022/06/26 21:48:19 rvm_ruby_string=ruby-2.7.1
    2022/06/26 21:48:19 GIMME_GO_VERSION=1.14.4
    2022/06/26 21:48:19 LOGNAME=buildbot
    2022/06/26 21:48:19 NETLIFY_NODE_VERSION=12.18.0
    2022/06/26 21:48:19 GOCACHE=/opt/buildhome/.gimme_cache/gocache
    2022/06/26 21:48:19 GEM_PATH=/opt/buildhome/.rvm/gems/ruby-2.7.1:/opt/buildhome/.rvm/gems/ruby-2.7.1@global
    2022/06/26 21:48:19 NVM_BIN=/opt/buildhome/.nvm/versions/node/v12.18.0/bin
    2022/06/26 21:48:19 GOPATH=/opt/buildhome/.gimme_cache/gopath
    2022/06/26 21:48:19 BINRC_VERSION=0.2.9
    2022/06/26 21:48:19 PANDOC_VERSION=2.4
    2022/06/26 21:48:19 rvm_delete_flag=0
    2022/06/26 21:48:19 SUDO_GID=0
    2022/06/26 21:48:19 GIMME_ENV_PREFIX=/opt/buildhome/.gimme/env
    2022/06/26 21:48:19 RUBY_VERSION=2.7.1
    2022/06/26 21:48:19 NETLIFY=true
    2022/06/26 21:48:19 _=/usr/bin/printenv
    2022/06/26 21:48:19 "23"
    2022/06/26 21:48:19 Finished
    2022/06/26 21:48:19 🚧 'wrangler pages <command>' is a beta command. Please report any issues to https://github.com/cloudflare/wrangler2/issues/new/choose
    2022/06/26 21:48:19 Compiled Worker successfully.
    2022/06/26 21:48:19 Validating asset output directory
    2022/06/26 21:48:19 Deploying your site to Cloudflare's global network...
    2022/06/26 21:48:19 Success: Assets published!
    2022/06/26 21:48:19 Success: Your site was deployed!
  ```
as expected!

Test case 2 (with size param):
<details>
  <summary>Code</summary>
  
  ```go
	package main

	import (
		"context"
		"log"
		"os"

		"github.com/cloudflare/cloudflare-go"
	)

	func main() {
		api, err := cloudflare.New(os.Getenv("API_KEY"), os.Getenv("EMAIL"))
		if err != nil {
			log.Fatal(err)
		}

		log.Println("== TESTING ON STAGING ==")
		api.BaseURL = "https://api.staging.cloudflare.com/client/v4"

		ctx := context.Background()

		logs, err := api.GetPagesDeploymentLogs(ctx, cloudflare.GetPagesDeploymentLogsParams{
			AccountID:    os.Getenv("ACCOUNT_TAG"),
			ProjectName:  "pages-testing-2",
			DeploymentID: "07deebfd-416c-459a-8586-82b063c2686c",

			SizeOptions: cloudflare.SizeOptions{
				Size: 3,
			},
		})

		if err != nil {
			log.Fatal(err)
		}

		for _, line := range logs.Data {
			log.Println(line.Line)
		}
	}
  ```
</details>

```
2022/06/26 22:14:19 == TESTING ON STAGING ==
2022/06/26 22:14:22 Cloning repository...
2022/06/26 22:14:22 From https://github.com/Walshy-testing/pages-testing
2022/06/26 22:14:22  * branch            328c4173ad737f36ea9930eefc083ca5f2bddcd3 -> FETCH_HEAD
```
works!

---

I did find a bug with test case 2 there, we weren't actually using the size param. Got a fix for that out but as you know can't push it live right now. Will let you decide if you want to merge or hold off until that fix is live.

Appreciate the help as always :)

File: pages_deployment.go [hunk 3]
```diff
@@ -153,22 +193,30 @@ func (api *API) GetPagesDeploymentInfo(ctx context.Context, params GetPagesDeplo
...
-	if params.Before > 0 {
-		v.Set("before", strconv.Itoa(params.Before))
-	} else if params.After > 0 {
-		v.Set("after", strconv.Itoa(params.After))
+
+	if params.ProjectName == "" {
+		return PagesDeploymentStageLogs{}, ErrMissingProjectName
+	}
+
+	if params.ProjectName == "" {
```
Comment (jacobbednarz) [inline pages_deployment.go:207]:
```suggestion
	if params.DeploymentID == "" {
```

File: pages_deployment.go [hunk 4]
```diff
@@ -181,10 +229,55 @@ func (api *API) GetPagesDeploymentStageLogs(ctx context.Context, params GetPages
...
+func (api *API) GetPagesDeploymentLogs(ctx context.Context, params GetPagesDeploymentLogsParams) (PagesDeploymentLogs, error) {
+	if params.AccountID == "" {
+		return PagesDeploymentLogs{}, ErrMissingAccountID
+	}
+
+	if params.ProjectName == "" {
+		return PagesDeploymentLogs{}, ErrMissingProjectName
+	}
+
+	if params.ProjectName == "" {
```
Comment (jacobbednarz) [inline pages_deployment.go:244]:
```suggestion
	if params.DeploymentID == "" {
```

File: pages_deployment.go [hunk 5]
```diff
@@ -181,10 +229,55 @@ func (api *API) GetPagesDeploymentStageLogs(ctx context.Context, params GetPages
...
 func (api *API) DeletePagesDeployment(ctx context.Context, params DeletePagesDeploymentParams) error {
+	if params.AccountID == "" {
+		return ErrMissingAccountID
+	}
+
+	if params.ProjectName == "" {
+		return ErrMissingProjectName
+	}
+
+	if params.ProjectName == "" {
```
Comment (jacobbednarz) [inline pages_deployment.go:277]:
```suggestion
	if params.DeploymentID == "" {
```

File: pages_deployment.go [hunk 6]
```diff
@@ -216,6 +317,18 @@ func (api *API) CreatePagesDeployment(ctx context.Context, params CreatePagesDep
...
 func (api *API) RetryPagesDeployment(ctx context.Context, params RetryPagesDeploymentParams) (PagesProjectDeployment, error) {
+	if params.AccountID == "" {
+		return PagesProjectDeployment{}, ErrMissingAccountID
+	}
+
+	if params.ProjectName == "" {
+		return PagesProjectDeployment{}, ErrMissingProjectName
+	}
+
+	if params.ProjectName == "" {
```
Comment (jacobbednarz) [inline pages_deployment.go:328]:
```suggestion
	if params.DeploymentID == "" {
```

File: pages_deployment.go [hunk 7]
```diff
@@ -234,6 +347,18 @@ func (api *API) RetryPagesDeployment(ctx context.Context, params RetryPagesDeplo
...
 func (api *API) RollbackPagesDeployment(ctx context.Context, params RollbackPagesDeploymentParams) (PagesProjectDeployment, error) {
+	if params.AccountID == "" {
+		return PagesProjectDeployment{}, ErrMissingAccountID
+	}
+
+	if params.ProjectName == "" {
+		return PagesProjectDeployment{}, ErrMissingProjectName
+	}
+
+	if params.ProjectName == "" {
```
Comment (jacobbednarz) [inline pages_deployment.go:358]:
```suggestion
	if params.DeploymentID == "" {
```

File: pages_deployment.go [hunk 8]
```diff
@@ -136,6 +164,18 @@ func (api *API) ListPagesDeployments(ctx context.Context, params ListPagesDeploy
...
 func (api *API) GetPagesDeploymentInfo(ctx context.Context, params GetPagesDeploymentInfoParams) (PagesProjectDeployment, error) {
+	if params.AccountID == "" {
+		return PagesProjectDeployment{}, ErrMissingAccountID
+	}
+
+	if params.ProjectName == "" {
+		return PagesProjectDeployment{}, ErrMissingProjectName
+	}
+
+	if params.ProjectName == "" {
```
Comment (jacobbednarz) [inline pages_deployment.go:175]:
```suggestion
	if params.DeploymentID == "" {
```

**Review by jacobbednarz — COMMENTED** [review]
just some copy/pasta to address and we're good!

File: pages_deployment.go (hunk 3 above)
Comment (jacobbednarz) [inline pages_deployment.go:207]:
```suggestion
	if params.DeploymentID == "" {
```

**Review by jacobbednarz — APPROVED** [review]

**Comment by jacobbednarz** [comment]
thanks for wrapping this up! 💯

**Pull request merged**